### PR TITLE
auto classify for each image

### PR DESCRIPTION
**Pull request by jackyzha0**

### CODE REVIEW

File: src/background.js [hunk 1]
```diff
@@ -26,28 +26,36 @@ const MOBILENET_MODEL_TFHUB_URL =
...
- *  Here it takes the url of the right-clicked image and the current tabId
- *  and forwards it to the imageClassifier's analyzeImage method.
- */
-function clickMenuCallback(info, tab) {
-  imageClassifier.analyzeImage(info.srcUrl, tab.id);
+
+function clickMenuCallback(_, tab) {
+  chrome.tabs.executeScript(tab.id, {
+    code: `var images = []
+           for (var i = 0; i < document.images.length; i++) {
```
Comment (ianmah) [inline src/background.js:38]: ever heard of a map??

File: src/background.js [hunk 2]
```diff
@@ -84,40 +92,34 @@ class ImageClassifier {
...
+  async analyzeImage(url) {
+    chrome.tabs.query({ active: true }, (tabs) => {
+      const tabId = tabs[0].id
+      if (!tabId) {
+        console.error('No tab.  No prediction.');
+        return;
+      }
+      if (!this.model) {
+        console.log('Waiting for model to load...');
+        setTimeout(() => { this.analyzeImage(url) }, FIVE_SECONDS_IN_MS);
```
Comment (ianmah) [inline src/background.js:104]: wtf is this can u pls make into a promise

File: src/background.js (hunk 1 above)
Comment (jackyzha0) [reply src/background.js:38]: no

File: src/background.js [hunk 3]
```diff
@@ -84,40 +92,34 @@ class ImageClassifier {
...
         async (img) => {
           if (!img) {
             console.error(
-                'Could not load image.  Either too small or unavailable.');
+              'Could not load image.  Either too small or unavailable.');
             return;
           }
           const predictions = await this.predict(img);
-          message = {action: 'IMAGE_CLICK_PROCESSED', url, predictions};
+          message = { action: 'IMAGE_CLICK_PROCESSED', url, predictions };
```
Comment (ianmah) [inline src/background.js:116]: why u send action if u dont check action later huh

File: src/background.js (hunk 2 above)
Comment (jackyzha0) [reply src/background.js:104]: dude this was here before lmao

File: src/background.js [hunk 4]
```diff
@@ -26,28 +26,36 @@ const MOBILENET_MODEL_TFHUB_URL =
...
+  onclick: clickMenuCallback,
+});
+
+chrome.runtime.onMessage.addListener((message) => {
+  if (message.method == "downloadImages") {
+    const allImages = []
+    message.images.forEach(function (v) {
+      allImages.push(v);
+    });
+    allImages.forEach(image => imageClassifier.analyzeImage(image))
```
Comment (ianmah) [inline src/background.js:57]: could this not just be `message.images.forEach(xxxx)`

Comment (jackyzha0) [reply src/background.js:57]: brain no work good

File: src/background.js (hunk 2 above)
Comment (ianmah) [reply src/background.js:104]: Ye LOOL it's fine

**Pull request merged**